src/ModelContextProtocol/Client/McpClientFactory.cs

```diff
@@ -25,13 +29,26 @@ public static class McpClientFactory
     /// <exception cref="InvalidOperationException"><paramref name="createTransportFunc"/> returns an invalid transport.</exception>
     public static async Task<IMcpClient> CreateAsync(
         McpServerConfig serverConfig,
-        McpClientOptions clientOptions,
+        McpClientOptions? clientOptions = null,
         Func<McpServerConfig, ILoggerFactory?, IClientTransport>? createTransportFunc = null,
         ILoggerFactory? loggerFactory = null,
         CancellationToken cancellationToken = default)
     {
         Throw.IfNull(serverConfig);
-        Throw.IfNull(clientOptions);
+
+        if (clientOptions is null)
+        {
+            var asmName = (Assembly.GetEntryAssembly() ?? Assembly.GetCallingAssembly()).GetName();
+
+            clientOptions ??= new()
```

Comment:
I think we could just cache a default instance -- it's immutable as well so it shouldn't create any problems.